hosting/src/components/AdminComponent.jsx

```diff
@@ -13,27 +13,41 @@ const AdminComponent = () => {
   const [outcomeTeam2, setOutcomeTeam2] = useState("");
   const [outcomeTeam3, setOutcomeTeam3] = useState("");
   const [isValid, setIsValid] = useState(false);
+  const [players, setPlayers] = useState([]);
 
-  const fetchTeams = async () => {
-    setIsLoading(true);
-    const response = await fetch("/api/competition/teams");
+  const getData = async (url, setFunc) => {
+    const response = await fetch(url);
     const json = await response.json();
-    setIsLoading(false);
 
     if (!response.ok) {
       throw new Error(json);
     }
 
-    setTeams(json);
+    setFunc(json);
+  };
+
+  const loadData = async () => {
+    setIsLoading(true);
+
+    try {
+      await Promise.all([
+        getData(`/api/players`, setPlayers),
+        getData(`/api/competition/teams`, setTeams),
+      ]);
+    } catch (err) {
+      console.log(err);
+    }
+
+    setIsLoading(false);
   };
 
   useEffect(() => {
-    fetchTeams();
+    loadData();
   }, []);
 
   useEffect(() => {
     selectTeams();
-  }, [teams]);
+  }, [teams, players]);
```

Comment:
I think this is why it loads 3 times - once from the `onSubmit` handler, once when teams loads and once when players loads.
I think you can get rid of this `useEffect` and move the `selectTeams` call from `onSubmit` into `loadData` after teams and players have loaded

Reply:
It looks like that without this `useEffect` it will choose all teams to be Argentina, assuming because `loadData` calling `selectedTeams` rerender the component; I did try to remove `teams` from the `useEffect` and it seems to work as intended

Reply:
I think if you only have `players` in the useEffect it might use a stale version of `teams`... which probably works fine  as that list shouldn't change but isn't ideal. I think if you remove `selectTeams` from the `onSubmit` handler, and change the `loadData` flow to look like this it should work

```
  const getData = async (url) => {
    const response = await fetch(url);
    const json = await response.json();

    if (!response.ok) {
      throw new Error(json);
    }

    return json;
  };

  const loadData = async () => {
    setIsLoading(true);

    try {
      const [players, teams] = await Promise.all([
        getData(`/api/players`),
        getData(`/api/competition/teams`),
      ]);

      setPlayers(players);
      setTeams(teams);
    } catch (err) {
      console.log(err);
    }

    setIsLoading(false);
  };
```